src/Transport/Receiving/MessagePump.cs

```diff
@@ -115,6 +115,29 @@ public Task StartReceive(CancellationToken cancellationToken = default)
             return Task.CompletedTask;
         }
 
+        public async Task ChangeConcurrency(PushRuntimeSettings newLimitations, CancellationToken cancellationToken = new CancellationToken())
+        {
+            var oldLimiter = concurrencyLimiter;
+            var oldMaxConcurrency = maxConcurrency;
+            concurrencyLimiter = new SemaphoreSlim(newLimitations.MaxConcurrency);
```

Comment:
Should the initial and maximum number of requests be specified, similar to [here](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/453/files#diff-bf785c1b37e39481e89856ea4195ca5c7e21f862b8eb2ffc43b63208a3c582e5R105)?